src/runtime/manifest.ts

```diff
@@ -113,14 +124,16 @@ export interface ManifestRequiredAPI {
 }
 
 /**
- * An definition of a function deployment as appears in the Manifest.
+ * A definition of a function/extension deployment as appears in the Manifest.
+ *
  * @alpha
  */
 export interface ManifestStack {
   specVersion: "v1alpha1";
   params?: WireParamSpec<any>[];
   requiredAPIs: ManifestRequiredAPI[];
   endpoints: Record<string, ManifestEndpoint>;
+  extensions: Record<string, ManifestExtension>;
```

Comment:
same here - make extensions undefined-able (`extensions?`)

Reply:
Same as above. It's already backwards compatible etc. If extensions are made undefined-able then the same should be true for endpoints... I don't want people thinking "We always have functions, but only sometimes have extensions" because it's not true

Reply:
Understand your point, but I see extensions use case for the Functions SDK as an "extensions" to the Functions SDK, not the core experience of using the Functions SDK.

For instance, both go and dart implementation for "Firebase Functions SDK" relied on writing on the functions.yaml by hand instead. In those cases, the functions.yaml definition we defined in the Node.js served as the schema definition.

By making extensions a "required" field, we would be making these functions.yaml to add:

```
extensions: {}
```

I don't like that experience since it would be surprising to have to write `extensions` on their functions manifest.

(Granted, we don't really expect developers to write functions.yaml by hand, but we are talking about schema definition, and I don't think `extensions` should be necessary)

Reply:
I agree that it's very surprising to have extensions mixed in with the functions. I made that argument strenuously and Thomas insisted that they must be. And since it must be, then we should be accurate and have extensions mixed in with the functions everywhere like he wanted. Especially since we might also support Go and dart for extensions at some point.